activerecord/test/cases/reflection_test.rb

```diff
@@ -328,7 +328,7 @@ def test_nested?
   def test_association_primary_key
     # Normal association
     assert_equal "id",   Author.reflect_on_association(:posts).association_primary_key.to_s
-    assert_equal "name", Author.reflect_on_association(:essay).association_primary_key.to_s
+    assert_equal "id",   Author.reflect_on_association(:essay).association_primary_key.to_s
```

Comment:
`association_primary_key` should return the association's primary key (i.e. `essay`'s primary key), but `name` is the column on `Author`.